File: csharp+vs+nunit/CarPlant/CarPlant.Tests/ToyotaCorollaSportsTest.cs
using NUnit.Framework;
using System;

namespace CarPlant
{
	[TestFixture]
	public class ToyotaCorollaSportsTest
	{

		private Car corolla;

		[SetUp]
		public void Init()
		{
			CarPlant.LOG = "";
			corolla = CarPlant.makeToyotaCorollaSports();
		}

		[Test]
		public void HasTheCorrectName()
		{
			Assert.AreEqual("Toyota Corolla sports", corolla.Name());
		}

		[Test]
		public void TurningLeftOnlyTurnsTheFrontWheels()
		{
			int degrees = TestHelpers.randomDegrees();
			corolla.TurnLeft(degrees);
			Assert.AreEqual("front left turning left " + degrees + " degrees\n"
			             + "front right turning left " + degrees + " degrees\n", CarPlant.LOG);
		}

		[Test]
		public void TurningRightOnlyTurnsTheFrontWheels()
		{
			int degrees = TestHelpers.randomDegrees();
			corolla.TurnRight(degrees);
			Assert.AreEqual("front left turning right " + degrees + " degrees\n"
			             + "front right turning right " + degrees + " degrees\n", CarPlant.LOG);
		}

		[Test]
		public void AcceleratingOnlyTurnsTheRearWheels()
		{
			int speed = TestHelpers.randomSpeed();
			corolla.Accelerate(speed);
			Assert.AreEqual("rear left accelerating " + speed + " kph\n"
			             + "rear right accelerating " + speed + " kph\n", CarPlant.LOG);
		}

	}
}

File: java/src/com/xpsurgery/carplanttests/ToyotaCorollaSportsTest.java
package com.xpsurgery.carplanttests;

import static org.junit.Assert.assertEquals;
import org.junit.Before;
import org.junit.Test;
import java.io.StringWriter;
import com.xpsurgery.carplant.CarPlant;
import com.xpsurgery.carplant.Car;

public class ToyotaCorollaSportsTest {

	private Car corolla;
  private StringWriter log;

	@Before
	public void setUp() throws Exception {
    log = new StringWriter();
		corolla = CarPlant.makeToyotaCorollaSports(log);
	}

	@Test
	public void hasTheCorrectName() {
		assertEquals("Toyota Corolla sports", corolla.name());
	}

	@Test
	public void turningLeftOnlyTurnsTheFrontWheels() {
		int degrees = TestHelpers.randomDegrees();
		corolla.turnLeft(degrees);
		assertEquals("front left turning left " + degrees + " degrees\n"
				   + "front right turning left " + degrees + " degrees\n", log.toString());
	}

	@Test
	public void turningRightOnlyTurnsTheFrontWheels() {
		int degrees = TestHelpers.randomDegrees();
		corolla.turnRight(degrees);
		assertEquals("front left turning right " + degrees + " degrees\n"
				   + "front right turning right " + degrees + " degrees\n", log.toString());
	}

	@Test
	public void acceleratingOnlyTurnsTheRearWheels() {
		int speed = TestHelpers.randomSpeed();
		corolla.accelerate(speed);
		assertEquals("rear left accelerating " + speed + " kph\n"
				   + "rear right accelerating " + speed + " kph\n", log.toString());
	}

}

